core/drivers/stpmic1.c

```diff
@@ -0,0 +1,773 @@
+		.low_power_reg	= BUCK3_PWRCTRL_REG,
+		.pull_down_reg	= BUCK_PULL_DOWN_REG,
+		.pull_down_pos	= BUCK3_PULL_DOWN_SHIFT,
+		.mask_reset_reg = MASK_RESET_BUCK_REG,
+		.mask_reset_pos = BUCK3_MASK_RESET_SHIFT,
+	},
+	{
+		.dt_node_name	= "buck4",
+		.voltage_table	= buck4_voltage_table,
+		.voltage_table_size = ARRAY_SIZE(buck4_voltage_table),
+		.control_reg	= BUCK4_CONTROL_REG,
+		.low_power_reg	= BUCK4_PWRCTRL_REG,
+		.pull_down_reg	= BUCK_PULL_DOWN_REG,
+		.pull_down_pos	= BUCK4_PULL_DOWN_SHIFT,
+		.mask_reset_reg = MASK_RESET_BUCK_REG,
+		.mask_reset_pos = BUCK4_MASK_RESET_SHIFT,
+	},
+	{
+		.dt_node_name	= "ldo1",
+		.voltage_table	= ldo1_voltage_table,
+		.voltage_table_size = ARRAY_SIZE(ldo1_voltage_table),
+		.control_reg	= LDO1_CONTROL_REG,
+		.low_power_reg	= LDO1_PWRCTRL_REG,
+		.mask_reset_reg = MASK_RESET_LDO_REG,
+		.mask_reset_pos = LDO1_MASK_RESET_SHIFT,
+	},
+	{
+		.dt_node_name	= "ldo2",
+		.voltage_table	= ldo2_voltage_table,
+		.voltage_table_size = ARRAY_SIZE(ldo2_voltage_table),
+		.control_reg	= LDO2_CONTROL_REG,
+		.low_power_reg	= LDO2_PWRCTRL_REG,
+		.mask_reset_reg = MASK_RESET_LDO_REG,
+		.mask_reset_pos = LDO2_MASK_RESET_SHIFT,
+	},
+	{
+		.dt_node_name	= "ldo3",
+		.voltage_table	= ldo3_voltage_table,
+		.voltage_table_size = ARRAY_SIZE(ldo3_voltage_table),
+		.control_reg	= LDO3_CONTROL_REG,
+		.low_power_reg	= LDO3_PWRCTRL_REG,
+		.mask_reset_reg = MASK_RESET_LDO_REG,
+		.mask_reset_pos = LDO3_MASK_RESET_SHIFT,
+	},
+	{
+		.dt_node_name	= "ldo4",
+		.voltage_table	= ldo4_voltage_table,
+		.voltage_table_size = ARRAY_SIZE(ldo4_voltage_table),
+		.control_reg	= LDO4_CONTROL_REG,
+		.low_power_reg	= LDO4_PWRCTRL_REG,
+		.mask_reset_reg = MASK_RESET_LDO_REG,
+		.mask_reset_pos = LDO4_MASK_RESET_SHIFT,
+	},
+	{
+		.dt_node_name	= "ldo5",
+		.voltage_table	= ldo5_voltage_table,
+		.voltage_table_size = ARRAY_SIZE(ldo5_voltage_table),
+		.control_reg	= LDO5_CONTROL_REG,
+		.low_power_reg	= LDO5_PWRCTRL_REG,
+		.mask_reset_reg = MASK_RESET_LDO_REG,
+		.mask_reset_pos = LDO5_MASK_RESET_SHIFT,
+	},
+	{
+		.dt_node_name	= "ldo6",
+		.voltage_table	= ldo6_voltage_table,
+		.voltage_table_size = ARRAY_SIZE(ldo6_voltage_table),
+		.control_reg	= LDO6_CONTROL_REG,
+		.low_power_reg	= LDO6_PWRCTRL_REG,
+		.mask_reset_reg = MASK_RESET_LDO_REG,
+		.mask_reset_pos = LDO6_MASK_RESET_SHIFT,
+	},
+	{
+		.dt_node_name	= "vref_ddr",
+		.voltage_table	= vref_ddr_voltage_table,
+		.voltage_table_size = ARRAY_SIZE(vref_ddr_voltage_table),
+		.control_reg	= VREF_DDR_CONTROL_REG,
+		.low_power_reg	= VREF_DDR_PWRCTRL_REG,
+		.mask_reset_reg = MASK_RESET_LDO_REG,
+		.mask_reset_pos = VREF_DDR_MASK_RESET_SHIFT,
+	},
+	{
+		.dt_node_name = "boost",
+	},
+	{
+		.dt_node_name = "pwr_sw1",
+	},
+	{
+		.dt_node_name = "pwr_sw2",
+	},
+};
+
+#define MAX_REGUL	ARRAY_SIZE(regulators_table)
+
+static const struct regul_struct *get_regulator_data(const char *name)
+{
+	unsigned int i = 0;
+
+	for (i = 0 ; i < MAX_REGUL ; i++)
```

Comment:
No space before `;` please.
I would have preferred writing `ARRAY_SIZE(regulators_table)` directly instead of using `MAX_REGUL` since you're indexing in the `regulators_table` just below.
￼